### PR TITLE
Fix/react version backwards compatibility

### PR DESCRIPTION
**Pull request by ndelangen**
Issue: #4191 

## What I did
- Ensure usage of react installed by user inside preview
- FIX compatibility issue with addon-info (it cannot use @storybook/components)

## How to test

I changed the versions of `package.json` the cra example, and verified it would work, then changed it back.

See: https://5c2f898b5f9f740009255238--storybooks-cra.netlify.com/?selectedKind=Button&selectedStory=with%20new%20info&full=0&addons=1&stories=1&panelRight=1&addonPanel=storybook%2Factions%2Factions-panel

---

most changes are related to addon-info

### CODE REVIEW

**Review by tmeasday — COMMENTED** [review]
I don’t get it. Didn’t we already make this change in a different pr? (To iframe.webpack.js).

**Comment by ndelangen** [comment]
@tmeasday indeed, I'm confused as well, where did this change go we worked on..

File: addons/info/src/components/PropTable.js [hunk 1]
```diff
@@ -1,10 +1,16 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 
-import { Table, Td, Th } from '@storybook/components';
 import PropVal from './PropVal';
 import PrettyPropType from './types/PrettyPropType';
 
+const Table = props => <table style={{}} {...props} />;
```
Comment (Hypnosphi) [inline addons/info/src/components/PropTable.js]: what's the point of adding an empty `style` object?

Comment (ndelangen) [reply addons/info/src/components/PropTable.js]: right nothing, doesn't output anything either though

Comment (Hypnosphi) [reply addons/info/src/components/PropTable.js]: Why not just use `table` then?

**Comment by shilman** [comment]
@ndelangen @tmeasday I'm confused about the status of these changes.
 - I checked out this branch and downgraded `react/react-dom` in `examples/cra-kitchen-sink/package.json` and ran storybook successfully
 - I checked out the `next` branch and did the same thing and ran `yarn storybook` and expected it to fail, but it succeeded.

@ndelangen Do we have a failing example in the repo that I can test to verify the fix?

**Comment by ndelangen** [comment]
> I checked out the next branch and did the same thing and ran yarn storybook and expected it to fail, but it succeeded.

You possibly forgot to regenerate the dll?

**Comment by ndelangen** [comment]
> Also, what's up with "part 1"? Are there more parts coming?

just ignore, I was expecting to need to experiment a lot, since I thought @tmeasday and me already fixed this.

File: addons/info/src/components/PropTable.js (hunk 1 above)
Comment (ndelangen) [reply addons/info/src/components/PropTable.js]: I don't feel strongly about this, but I like the look of jsx when it's just components, vs components & html mixed.

I don't think there are any pros or cons to speak of really.

**Comment by shilman** [comment]
@ndelangen I did `yarn bootstrap --reset --core` and it regenerated the DLLs

File: addons/info/src/components/PropTable.js (hunk 1 above)
Comment (Hypnosphi) [reply addons/info/src/components/PropTable.js]: One con is that it makes code less understandable at a first glance. If you really want camelcase, you can do this:

```
const Table = 'table'
```

**Comment by shilman** [comment]
I tried to test this out in the Wix example by creating build-packs and modifying their package.json as follows:

```
    "@storybook/addon-links": "file:/Users/shilman/projects/storybook/new/storybook/packs/storybook-addon-links.tgz",
    "@storybook/addon-options": "file:/Users/shilman/projects/storybook/new/storybook/packs/storybook-addon-options.tgz",
    "@storybook/react": "file:/Users/shilman/projects/storybook/new/storybook/packs/storybook-react.tgz",
    "@storybook/core": "file:/Users/shilman/projects/storybook/new/storybook/packs/storybook-core.tgz",
    "@storybook/ui": "file:/Users/shilman/projects/storybook/new/storybook/packs/storybook-ui.tgz",
    "@storybook/addons": "file:/Users/shilman/projects/storybook/new/storybook/packs/storybook-addons.tgz",
    "@storybook/components": "file:/Users/shilman/projects/storybook/new/storybook/packs/storybook-components.tgz",
```

I'm still getting the `TypeError: Object(...) is not a function` repro after doing this, though I'm not entirely sure whether I'm doing it properly.

**Comment by shilman** [comment]
@ndelangen I've released this as `4.1.5-react-debug.0` and confirmed that it fixes the issue with `wix-style-react`. I tried a simpler fix proposed by @tmeasday that just cherry-picks https://github.com/storybooks/storybook/commit/f7c325bd5f19b6479122a07ce3dbff121aaf7285 onto `master` but I'm getting the following error in `createDlls`. I suspect that there's some simple explanation, but unless I hear back soon I'm just going to merge and release this PR. I'll also take care of the `addon-links` issue that Tom mentioned before releasing.

```
ERROR in /Users/shilman/projects/storybook/new/storybook/node_modules/@storybook/react-komposer/dist/compose.js
Module not found: Error: Can't resolve '@storybook/react-stubber' in '/Users/shilman/projects/storybook/new/storybook/node_modules/@storybook/react-komposer/dist'
 @ /Users/shilman/projects/storybook/new/storybook/node_modules/@storybook/react-komposer/dist/compose.js 45:20-55
 @ /Users/shilman/projects/storybook/new/storybook/node_modules/@storybook/react-komposer/dist/index.js
 @ ./dist/compose.js
 @ ./dist/index.js
 @ dll storybook_ui

ERROR in ./dist/modules/ui/components/search_box.js
Module not found: Error: Can't resolve 'react-fuzzy' in '/Users/shilman/projects/storybook/new/storybook/lib/ui/dist/modules/ui/components'
 @ ./dist/modules/ui/components/search_box.js 36:41-63
 @ ./dist/modules/ui/containers/search_box.js
 @ ./dist/modules/ui/routes.js
 @ ./dist/modules/ui/index.js
 @ ./dist/index.js
 @ dll storybook_ui
```

**Review by shilman — APPROVED** [review]
Approving this because I've verified that it works--not because I understand the solution or the alternatives. I've also fixed `addon-links` in an ugly way and have verified the fix.

**Comment by shilman** [comment]
@Hypnosphi The typescript CRA example Teamcity config has been merged and is now being applied to the master branch, which doesn't contain the TS CRA example. I'm ignoring the build error and merging, but if you have any suggestions on a better solution for this (cherry-pick the PR into `master`? OR ... ?) any ideas are appreciated. Thanks!

**Comment by Hypnosphi** [comment]
@shilman You just need to pass `false` as third argument to skip the actual test, see e.g. `PREACT` https://github.com/storybooks/storybook/blob/master/.teamcity/OpenSourceProjects_Storybook/buildTypes/StorybookApp.kt#L20

**Pull request merged**